### DISCUSSION TITLE
Request :: type property

### DISCUSSION THREAD

**Discussion by jecoute** [mozts2005/ZendeskApi_v2#533] [Features]
In the API, type is set as an enumeration, but the Zendesk API has this property as a String field. 

![image](https://user-images.githubusercontent.com/1268162/136642045-30603ac3-e29a-4635-bb1f-1def979f16cf.png)

![image](https://user-images.githubusercontent.com/1268162/136642068-91f1c2fc-83f3-4dd4-91dc-142a7716e8c7.png)

**Comment by mozts2005** [comment]
Yes, the enum is used to limit the options to a fixed list the converter uses the names not the values.

Do we need to add an option to the enum?

**Reply by jecoute** [reply]
Hmm...  I found when i used the enum, I didn't get a value in the request for it after it was created.

**Reply by mozts2005** [reply]
We have a number of tests that confirm this functionality. Can you please create a PR with a test that replicates your issue?

https://github.com/Speedygeek/ZendeskApi_v2/blob/909b7bbea666aa863331a84f698c0b33da40dada/tests/ZendeskApi_v2.Tests/RequestTests.cs#L178

**Comment by jecoute** [comment]
I'm not sure what's happening, but for some reason I wasn't able to make any PR.   Attached you can see the screenshots where we use your zendesk client.    From the first screenshot, UnitTest1, I've set the Type to 'task'.   The second one, UnitTest2,shows the response received and the Type is defaulted to 'incident'.   

I think it probably should be an enum where we can at cast the enum with ToString() to get the string literal of the enum to set the desired type which the API is showing, see UnitTest 4.

![zendeskUnitTest1](https://user-images.githubusercontent.com/1268162/138978855-0023de53-a4fe-4aa4-a8cb-660417765438.png)
![zendeskUnitTest2](https://user-images.githubusercontent.com/1268162/138978860-bc1db29e-6b9d-4b6d-a976-cedd5da0b829.png)
![zendeskUnitTest3](https://user-images.githubusercontent.com/1268162/138978867-472caa2a-cf58-42bc-8d67-ed4dd2b74521.png)
![zendeskUnitTest4](https://user-images.githubusercontent.com/1268162/138978872-35f4118a-0dc4-4781-9f37-57a819b0fbac.png)

**Reply by mozts2005** [reply]
Based on what I am reading you are using your own instance of the Newtonsoft seriallizer and not the one configured by the library. 


try the following in a NUnit test class.  Please make sure you are using version 3.12.0. 

```csharp


        [Test]
        public void CanCreateRequests()
        {

            var api = new ZendeskApi(Organization.SiteURL, Admin.Email, Admin.Password);
            var req = new Request
            {
                Subject = "end user request test",
                Type = RequestType.Task,
                Comment = new Comment
                { Body = "end user test", HtmlBody = "end user test with </br> new line", Public = true },
                Requester = new Requester
                {
                    Name = "Test Name"
                },
                Tags = new List<string> { "tag1", "tag2" }
            };

            var res = api.Requests.CreateRequest(req);

                Assert.That(res, Is.Not.Null);
                Assert.That(res.Request, Is.Not.Null);
                Assert.That(res.Request.Id.HasValue, Is.True);
                Assert.That(res.Request.Type, Is.EqualTo(RequestType.Task));
        }
```

https://github.com/Speedygeek/ZendeskApi_v2/blob/7b9aa5afb8c7860c8947a55c7a7ce6753d76efb7/src/ZendeskApi_v2/Models/Requests/Request.cs#L36-L38

As you can see, we are using the StringEnumConverter to make Newtonsoft seriallizer output a string. 


We would be happy to fix an issue once we have a reproducible failing test case. Given that only a fix set of strings allowed, it is incorrect to allow a generic string to be the property type.
